etcdserver/api/v3rpc/grpc.go

```diff
@@ -36,8 +36,11 @@ func Server(s *etcdserver.EtcdServer, tls *tls.Config) *grpc.Server {
 	}
 	opts = append(opts, grpc.UnaryInterceptor(newUnaryInterceptor(s)))
 	opts = append(opts, grpc.StreamInterceptor(newStreamInterceptor(s)))
-
+	// set MaxMsgSize be a bit higher than that of MaxRequestBytes bypasses grpc request size check
```

Comment:
that was what I was thinking. the code works as expected if `request size` < `(MaxRequestBytes + grpcOverheadBytes)`.

 If `request size` > `MaxRequestBytes` && `request size` < `(MaxRequestBytes + grpcOverheadBytes)`, client expects an `ErrRequestTooLarge` error. 

 If `request size` > `(MaxRequestBytes + grpcOverheadBytes)`, the client will receive an grpc error [`grpc: received message larger than max...`](https://github.com/grpc/grpc-go/blob/9f919f7b81bbce4fb49573291f5822c233d6054b/server.go) instead.

I wonder whether this behavior is okay. client might recv two different errors that mean pretty much the same thing.